### PR TITLE
feat: add listUsers and listUsersInGroup actions to iamAction

### PR DESCRIPTION
**Pull request by kita3222**
## Problem

#1375 

Added listUsers and listUsersInGroup actions to iamAction.

**Issue number, if available:**

## Changes

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .changeset/short-toes-whisper.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/backend-auth': patch
```
Comment (edwardfoyle) [inline .changeset/short-toes-whisper.md:2]:
```suggestion
'@aws-amplify/backend-auth': minor
```

**Review by edwardfoyle — DISMISSED** [review]
LGTM, just need to update the changeset

**Comment by kita3222** [comment]
@edwardfoyle 
Thanks for confirming! We've taken care of it.

**Review by awsluja — APPROVED** [review]

**Review by 0618 — DISMISSED** [review]

**Review by sobolk — APPROVED** [review]

**Comment by kita3222** [comment]
@edwardfoyle @sobolk @awsluja @0618 
The conflict was resolved. Please check again！

**Review by sobolk — APPROVED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**